### PR TITLE
Add hint re: fuzzer to api changes doc

### PR DESCRIPTION
**Pull request by pmorie**
@lavalamp

### CODE REVIEW

File: docs/devel/api_changes.md [hunk 1]
```diff
@@ -236,7 +236,8 @@ assumptions.  If you have added any fields which need very careful formatting
 "this slice will always have at least 1 element", you may get an error or even
 a panic from the `serialization_test`.  If so, look at the diff it produces (or
 the backtrace in case of a panic) and figure out what you forgot.  Encode that
-into the fuzzer's custom fuzz functions.
+into the fuzzer's custom fuzz functions.  Hint: if you added defaults for fields,
+those fields will need to be fuzzed.
```
Comment (rjnagal) [inline docs/devel/api_changes.md]: s/need to be fuzzed/need to be custom-handled/

Probably point to an example in /pkg/api/testing/fuzzer.go : eg. https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/testing/fuzzer.go#L232

**Comment by lavalamp** [comment]
Yeah... if a bad object makes you panic you should fix the thing that panics, not the fuzzer! We really need two fuzz tests-- one for testing serialization for valid objects, and one that tests that invalid objects don't make anything crash.

**Comment by pmorie** [comment]
My language is imprecise -- it should really be 'if a field is defaulted,
that field will need to have a custom handler that ensures that the field
is fuzzed to a non-nil value'.

**Comment by rjnagal** [comment]
That's what I wanted the text to be updated to :)

**Comment by pmorie** [comment]
@rjnagal Fixed

**Comment by smarterclayton** [comment]
Alternatively, we should really make the round trip test aware of defaults for return comparison.  That's a fairly trivial change to round tripper - it's just invoking the defaulting function for the type.

**Comment by rjnagal** [comment]
LGTM for the new documentation.

I like the idea of making defaulting function part of the round trip test. We can merge the doc change and I can work on the changing the round tripper.

**Comment by pmorie** [comment]
@rjnagal created #7337

**Pull request merged**